modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2868,13 +2909,19 @@ def _pick_axis(get_axis, sizes_cache):
         if axis == 0:
             # Pass shape caches instead of values in order to not trigger shape computation.
             new_index, new_row_lengths = _pick_axis(
-                self._get_index, self._row_lengths_cache
+                self.copy_index_cache, self._row_lengths_cache
```

Comment:
`.copy_index_cache()` doesn't trigger index materialization in comparison with `._get_index()` that does it